### PR TITLE
feat: flux cd app details

### PR DESCRIPTION
**Pull request by arunjaindev**
# Description
Feature: FluxCD app details

Fixes # (issue)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Checklist:

* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [x] I have performed a self-review of my own code
* [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

**Review by AbhishekA1509 — CHANGES_REQUESTED** [review]

**Review by AbhishekA1509 — CHANGES_REQUESTED** [review]

File: src/Pages/App/Details/ExternalFlux/ExternalFluxAppDetails.tsx [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+    const [isAppDetailsLoading, appDetailsResult, appDetailsError, reloadAppDetails] = useAsync(
+        () => getExternalFluxCDAppDetails(clusterId, namespace, appName, isKustomization),
+        [],
+        isSuperAdmin,
+        {
+            resetOnChange: false,
+        },
+    )
+
+    useInterval(reloadAppDetails, 30000)
```
[the remark this answers is not in the document]
Comment (arunjaindev) [reply src/Pages/App/Details/ExternalFlux/ExternalFluxAppDetails.tsx:55]: please confirm whether it should be setInterval or setTimeout ?

File: src/assets/icons/ic-argocd-app.svg [hunk 2]
```diff
@@ -1,44 +1,44 @@
 <svg width="24" height="24" viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg">
-<g clip-path="url(#clip0_6805_4500)">
+<g clip-path="url(#clip0_6967_6190)">
```
[the remark this answers is not in the document]
Comment (arunjaindev) [reply src/assets/icons/ic-argocd-app.svg:2]: will verify with main while merging

File: src/Pages/App/Details/ExternalFlux/types.tsx [hunk 3]
```diff
@@ -0,0 +1,6 @@
+export interface ExternalFluxAppDetailParams {
```
[the remark this answers is not in the document]
Comment (arunjaindev) [reply src/Pages/App/Details/ExternalFlux/types.tsx:1]: no interface exists of such kind as of now

File: src/components/v2/appDetails/sourceInfo/utils.tsx [hunk 4]
```diff
@@ -1,13 +1,15 @@
...
+import React from 'react'
 
 export const getEnvironmentName = (
-    appType: AppType,
     clusterName: string,
     namespace: string,
     environmentName: string,
-): string => {
-    if (appType === AppType.EXTERNAL_ARGO_APP) {
+): string | JSX.Element => {
```
[the remark this answers is not in the document]
Comment (arunjaindev) [reply src/components/v2/appDetails/sourceInfo/utils.tsx:7]: gives linting error

File: src/Pages/App/Details/ExternalFlux/utils.tsx [hunk 5]
```diff
@@ -0,0 +1,11 @@
+import { EXTERNAL_FLUX_APP_STATUS } from './types'
+
+export const getAppStatus = (appStatus: string): string => {
+    if (appStatus === 'True') {
```
Comment (vivek-devtron) [inline src/Pages/App/Details/ExternalFlux/utils.tsx:4]: Please add comments here

File: src/Pages/App/Details/ExternalFlux/ExternalFluxAppDetails.tsx [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+import AppDetailsComponent from '../../../../components/v2/appDetails/AppDetails.component'
+import { getAppStatus } from './utils'
+import { AppDetails } from '../../../../components/v2/appDetails/appDetails.type'
+
+const ExternalFluxAppDetails = () => {
+    const { clusterId, appName, namespace, templateType } = useParams<ExternalFluxAppDetailParams>()
+    const { isSuperAdmin } = useMainContext()
+    const isKustomization = templateType === FluxCDTemplateType.KUSTOMIZATION
+
+    let initTimer = null
```
Comment (vivek-devtron) [inline src/Pages/App/Details/ExternalFlux/ExternalFluxAppDetails.tsx:39]: This should be outside of the component

File: src/Pages/App/Details/ExternalFlux/ExternalFluxAppDetails.tsx [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+            initTimer = setTimeout(reloadAppDetails, 30000)
+            const genericAppDetail = {
+                ...appDetailsResult.result,
+                appStatus: getAppStatus(appDetailsResult.result.appStatus),
+            }
+            IndexStore.publishAppDetails(genericAppDetail, AppType.EXTERNAL_FLUX_APP)
+        }
+    }, [appDetailsResult])
+
+    const isLoading = isAppDetailsLoading && (!appDetailsResult || appDetailsError)
```
Comment (vivek-devtron) [inline src/Pages/App/Details/ExternalFlux/ExternalFluxAppDetails.tsx:68]: we should move this line at line no 77 and remove the `appDetailsError` condition

**Review by vivek-devtron — APPROVED** [review]

**Pull request merged**